src/ast/ast.h

```diff
@@ -973,6 +972,7 @@ class MapAddr : public Node {
   }
 
   Map *map = nullptr;
+  SizedType map_addr_type;
```

Comment:
Nit: since the class is called `MapAddr`, shouldn't this just be named `type`?

Reply:
We're a bit inconsistent with these member names. I was matching VariableAddr's `var_addr_type`.

Reply:
Ah it seems like maybe we don't name any of these `type` so as not to confused with the method `type`